charts/timescaledb-single/admin-guide.md

```diff
@@ -381,6 +383,109 @@ stanza: poddb
             backup reference list: 20190904-071709F
 ```
 
+## Callbacks
+Patroni will trigger some callbacks on certain events. These are:
+
+- on_reload
+- on_restart
+- on_role_change
+- on_start
+- on_stop
+
+If you wish to have *your* script run after a certain event happens, you can do so by configuring
+`callbacks.configMap` to point to a [ConfigMap](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#configmap-v1-core).
+This ConfigMap should exist before you install a chart. The data keys that match the event names will be executed
+if the event happens. For convenience the `all` key will be executed at every event.
+
+Inside these callbacks, you have access to the environment variables of the pod, except the `PATRONI_` environment variables.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example-patroni-callbacks
+data:
+  on_start:
+    #!/bin/bash
+    echo "I just started"
+  on_role_change:
+    #!/bin/bash
+    curl http://${MYAPP}/
+  all: |
+    #!/bin/bash
+    echo "$0 happened"
+```
+
+### Example: Register Patroni events in a table
+
+
+*example-patroni-callbacks.yaml*
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example-patroni-callbacks
+data:
+  all: |
+    #!/bin/bash
+
+    # This script should only run on a primary instance
+    [ "$2" != "master" ] && exit 0
```

Comment:
Wonder if we should be more explicit in the above paragraph that this is how you check for only running on the master? 

e.g., I don't want our current postStart to run on all nodes do I?

Reply:
> I don't want our current postStart to run on all nodes do I?

That's the tough thing, where and when should your script run (generically)? For the use case that we're adding this feature it's quite clear: On the master only, but it's not that clearcut.

By using the `callbacks` and leaving it up to the designer of the script we allow all options.

A few usecases:

1. a database should be created
2. Events need to be recorded
3. Users need to be created (these users may change over time)

- For 1) We could use `post_init.sh`, which is executed exactly once in the lifetime of a cluster.
- For 2) we need to execute for every event that happens
- For 3) We need to run on the master only

Arguably, 3) should actually be a Job that is created if a change happens to the list of Users.

Patroni calls the scripts with 3 arguments (see https://patroni.readthedocs.io/en/latest/SETTINGS.html#postgresql), which allow you to verify what role the current node has.

Documenting this may be best option here.

Reply:
Yeah I mainly meant documenting what the args (and their order) so people can figure it out